spanner/api/Program.cs

```diff
@@ -0,0 +1,989 @@
+                        {"SingerId", SpannerDbType.Int64},
+                        {"AlbumId", SpannerDbType.Int64},
+                        {"MarketingBudget", SpannerDbType.Int64},
+                    });
+                var cmdLookup =
+                    connection.CreateSelectCommand("SELECT * FROM Albums");
+                using (var reader = await cmdLookup.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        if (reader.GetInt64(0) == 1 && reader.GetInt64(1) == 1)
+                        {
+                            cmd.Parameters["SingerId"].Value =
+                                reader.GetInt64(0);
+                            cmd.Parameters["AlbumId"].Value =
+                                reader.GetInt64(1);
+                            cmd.Parameters["MarketingBudget"].Value = 100000;
+                            await cmd.ExecuteNonQueryAsync();
+                        }
+                        if (reader.GetInt64(0) == 2 && reader.GetInt64(1) == 2)
+                        {
+                            cmd.Parameters["SingerId"].Value =
+                                reader.GetInt64(0);
+                            cmd.Parameters["AlbumId"].Value =
+                                reader.GetInt64(1);
+                            cmd.Parameters["MarketingBudget"].Value = 500000;
+                            await cmd.ExecuteNonQueryAsync();
+                        }
+                    }
+                }
+            }
+            Console.WriteLine("Updated data.");
+            return;
+            // [END update_data]
+        }
+
+        public static async Task QueryNewColumnAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data_with_new_column]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to Cloud Spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd =
+                    connection.CreateSelectCommand("SELECT * FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId : {reader["SingerId"]} "
+                        + $"AlbumId : {reader["AlbumId"]} "
+                        + $"MarketingBudget : {reader["MarketingBudget"]}");
+                    }
+                }
+            }
+            return;
+            // [END query_data_with_new_column]
+        }
+
+        internal class CustomTransientErrorDetectionStrategy
+            : ITransientErrorDetectionStrategy
+        {
+            public bool IsTransient(Exception ex)

```

Comment:
you can shorten this by using this expression body.

public bool IsTransient(Exception ex) => ex.IsTransientSpannerFault();